src/config.js

```diff
@@ -29,6 +29,26 @@ const env = from(process.env, {
     asTextFileContent: (path) => getFileContent(path).toString().trim(),
 });
 
+const vaultAuthMethod = env.get('VAULT_AUTH_METHOD').required().asEnum(['K8S', 'APP_ROLE']);
+let vaultAuth;
+if (vaultAuthMethod === 'K8S') {
+    vaultAuth = {
+        k8s: {
+            token: env.get('VAULT_K8S_TOKEN_FILE').default('/var/run/secrets/kubernetes.io/serviceaccount/token').asTextFileContent(),
+            role: env.get('VAULT_K8S_ROLE').required().asString(),
+        },
+    };
+} else if (vaultAuthMethod === 'APP_ROLE') {
+    vaultAuth = {
+        appRole: {
+            // Generated per: https://www.vaultproject.io/docs/auth/approle#via-the-cli-1
+            // Or: https://github.com/kr1sp1n/node-vault/blob/70097269d35a58bb560b5290190093def96c87b1/example/auth_approle.js
+            roleId: env.get('VAULT_ROLE_ID_FILE').default('/vault/role-id').asTextFileContent(),
+            roleSecretId: env.get('VAULT_ROLE_SECRET_ID_FILE').default('/vault/role-secret-id').asTextFileContent(),
+        },
+    };
+}
```

Comment:
how about throwing if no valid vault auth method is provided? should the service be allowed to start under those conditions? I think probably not.

Reply:
it's not possible to start the service with the auth method other than specified in enum

Reply:
understood. thanks.